newdevtech/newdev_api
# #166276462 Add error handler/ notFoundRoute middleware

#### What does this PR do?
- adds a `notfound route` middleware
- adds a default `error handler`
- adds test for `notFoundRoute` middleware
- remove` `sampleTest.spec.js`
- updates test script in `package.json`
- adds `nyc` config to `package.json`

#### Description of Task to be completed?
- As a user, when I try to access a route that does not exist, I should receive an appropriate error message.
- We need an error handler to provide a uniform error message format that will be used throughout the application

#### How should this be manually tested?
- run `npm run dev`
- try to access a non-existent route, let's say `localhost:3000/api/v1/non-existent route`
- you should receive a response with message `Route not found`.

#### What are the relevant pivotal tracker stories?
[#166276462](https://www.pivotaltracker.com/story/show/166276462)
